devito/interfaces.py

```diff
@@ -127,7 +132,10 @@ def set_initializer(self, lambda_initializer):
         self.initializer = lambda_initializer
 
     def _allocate_memory(self):
-        self._data = aligned(np.zeros(self.shape, self.dtype, order='C'), alignment=64)
+        if self.memmap:
+            self._data = np.memmap(filename=(self.f), dtype=self.dtype, mode='w+', shape=tuple(self.shape), order='C')
```

Comment:
About the tuple(), it was from a previous commit, where at one place the type of the tuple received is defined from another package other than the default CPython tuple, which caused a type error and I had to cast it to tuple. I just removed it and it works fine now.